nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -74,9 +87,26 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
         self._reduced_lm_loss_buffer = []
         self._reduced_sop_loss_buffer = []
 
+        # build_model returns a list of modules which are used for interleaved pipeline parallelism
+        self.model = build_model(
+            model_provider_func=self.model_provider_func,
+            wrap_with_ddp=False,
+            virtual_pipeline_model_parallel_size=self.cfg.get('virtual_pipeline_model_parallel_size', None),
```

Comment:
Can we remove all instances of interleaving from this PR so that when a PR does add it, it is cleaner?

Reply:
Yes. I also removed all places where we have a conditional if isinstance(self.model, list): coz the model will never be a list unless we use interleaved right. So removed a bunch of code.